src/vmm/src/builder.rs

```diff
@@ -556,7 +565,22 @@ pub fn build_microvm_from_snapshot(
     .map_err(Error::SeccompFilters)
     .map_err(StartMicrovmError::Internal)?;
 
-    Ok(vmm)
+    {
+        let vmm = vmm_reference.lock().expect("Poisoned lock");
+
+        vm_resources.set_vm_config(&VmConfig {
+            vcpu_count: Some(vmm.vcpus_handles.len() as u8),
+            mem_size_mib: Some(mem_size_mib(vmm.guest_memory()) as usize),
+            ht_enabled: None,
+            cpu_template: None,
+            track_dirty_pages: vmm
+                .guest_memory()
+                .iter()
+                .any(|region| (region.bitmap().is_some()))
+        }).map_err(SetVmResources)?;
+    }
```

Comment:
I'd move this a bit higher up in the function body, and remove the extra code block. We can also use the `vmm` variable that is already in scope for this function